File: 581-Shortest Unsorted Continuous Subarray.py
class Solution:
    def findUnsortedSubarray(self, nums: 'List[int]') -> 'int':
        start, end, length = 0, 0, len(nums)
        nums_sorted = sorted(nums)
        for n in range(length):
            if nums[n] != nums_sorted[n]:
                start = n
                break
        for n in range(length - 1, -1, -1):
            if nums[n] != nums_sorted[n]:
                end = n
                break
        return end - start + 1 if end > start else 0

File: C++/581-Shortest_Unsorted_Continuous_Subarray.cpp
// Runtime: 32 ms, faster than 92.70% of C++ online submissions for Shortest Unsorted Continuous Subarray.
// Memory Usage: 10.4 MB, less than 100.00% of C++ online submissions for Shortest Unsorted Continuous Subarray.
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int l = -1, r = -2, mn = nums[nums.size() - 1], mx = nums[0];
        for (int i = 0; i < nums.size(); ++i) {
            mx = max(mx, nums[i]);
            if (nums[i] < mx)
                r = i;
            mn = min(mn, nums[nums.size() - i - 1]);
            if (mn < nums[nums.size() - i - 1])
                l = nums.size() - i - 1;
        }
        return r - l + 1;
    }
};
